src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,763 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+                if (param is not List<ObjectFieldNode> paramList)
+                {
+                    throw new DataApiBuilderException(message: $"Unsupported type used with {fieldName} in the create mutation input",
+                                                      statusCode: HttpStatusCode.BadRequest,
+                                                      subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                }
+
+                MultipleCreateStructure multipleCreateStructure = new(
+                    entityName: entityName,
+                    higherLevelEntityName: entityName,
+                    inputMutParams: parsedInput);
+
+                PerformDbInsertOperation(context, paramList, sqlMetadataProvider, multipleCreateStructure, nestingLevel: 0);
+
+                if (multipleCreateStructure.CurrentEntityCreatedValues is not null)
+                {
+                    primaryKeysOfCreatedItemsInTopLevelEntity.Add(FetchPrimaryKeyFieldValues(sqlMetadataProvider, entityName, multipleCreateStructure.CurrentEntityCreatedValues));
+                }
+            }
+
+            return primaryKeysOfCreatedItemsInTopLevelEntity;
+        }
+
+        /// <summary>
+        /// 1. Identifies the order of insertion into tables involved in the create mutation request.
+        /// 2. Builds and executes the necessary database queries to insert all the data into appropriate tables.
+        /// </summary>
+        /// <param name="context">Hotchocolate's context for the graphQL request.</param>
+        /// <param name="parameters">Mutation parameter arguments</param>
+        /// <param name="sqlMetadataProvider">SqlMetadataprovider for the given database type.</param>
+        /// <param name="multipleCreateStructure">Wrapper object for the current entity for performing the multiple create mutation operation</param>
+        /// <param name="nestingLevel">Current depth of nesting in the multiple-create request</param>
+        private void PerformDbInsertOperation(
+            IMiddlewareContext context,
+            object? parameters,
+            ISqlMetadataProvider sqlMetadataProvider,
+            MultipleCreateStructure multipleCreateStructure,
+            int nestingLevel)
+        {
+
+            if (multipleCreateStructure.InputMutParams is null || parameters is null)
+            {
+                throw new DataApiBuilderException(
+                        message: "The input for a multiple create mutation operation cannot be null.",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+            }
+
+            // For One - Many and Many - Many relationship types, the entire logic needs to be run for each element of the input.
+            // So, when the input parameters is of list type, we iterate over the list and run the logic for each element.
```

Comment:
unclear what you mean by "the entire logic"

Reply:
updated the comment